client/src/app/modules/events/event-edit/event-edit.component.ts

```diff
@@ -1,66 +1,169 @@
-import { IEventSchema } from '../../../interfaces';
+import { IEventSchema, IEventLocation } from '../../../interfaces';
 
 @Component({
   selector: 'app-event-edit',
   templateUrl: './event-edit.component.html',
   styleUrls: ['./event-edit.component.scss']
 })
 export class EventEditComponent implements OnInit {
-  event: IEventSchema;
+  // Careful with variable naming, 'event' is global var in ECMA Script
+  @Output() iEvent: IEventSchema;
   isCreate = false;
+  eventForm: FormGroup;
 
   constructor(
     private eventService: EventService,
     private router: Router,
     private route: ActivatedRoute
   ) {
-    this.event = new IEventSchema();
+    this.iEvent = new IEventSchema();
+    this.eventForm = this.getEventForm();
    }
 
   ngOnInit() {
-    if (this.route.snapshot.params['id'] === 'new') {
-      this.isCreate = true;
-    } else {
-      this.eventService.getEvent(this.route.snapshot.params['id'])
-        .subscribe((event) => {
-          this.event = event;
+    this.setIsCreateState();
+    if (!this.isCreate) {
+      this.getEvent().subscribe(
+        (receivedEvent) => {
+          this.iEvent = receivedEvent;
+          this.setEventFormValues(receivedEvent);
         },
-        (reason) => {
+        (error) => {
           this.router.navigate(['/']);
         }
       );
-      this.isCreate = false;
+    } else {
+      this.setEventFormValues(this.iEvent);
     }
+    this.setEventFormValidators();
   }
 
-  create() {
-    this.eventService.createEvent(this.event)
+  public create(): void {
+    this.updateEventSchema();
+    this.eventService.createEvent(this.iEvent)
       .subscribe(
         (event) => {
-          this.router.navigate(['/']); // success path
+          this.router.navigate(['/']);
         },
         (error) => {
           console.log('Following error appeared: ', error);
         }
       );
   }
 
-  edit() {
-    this.eventService.editEvent(this.route.snapshot.params['id'], this.event)
+  public edit(): void {
+    this.updateEventSchema();
+    this.eventService.editEvent(this.route.snapshot.params['id'], this.iEvent)
       .subscribe((event) => {
         this.router.navigate(['/']);
       },
-      (reason) => {
-        console.log('Following error appeared: ', reason);
-      }
-    );
+        (reason) => {
+          console.log('Following error appeared: ', reason);
+        }
+      );
   }
 
-  cancel() {
+  public cancel(): void {
     this.router.navigate(['/']);
   }
 
+  public getEvent(): Observable<IEventSchema> {
+    return this.eventService.getEvent(this.route.snapshot.params['id']);
+  }
+
+  private setIsCreateState(): void {
+    if (this.route.snapshot.params['id'] === 'new') {
+      this.isCreate = true;
+    } else {
+      this.isCreate = false;
+    }
+  }
+
+  /**
+   * Binding Event object to form model
+   */
+  private updateEventSchema(): void {
+    const toBoolean = (value: String): boolean =>  !value || value === 'false' ? false : true;
```

Comment:
This could be instead:
const toBoolean = (value: String): boolean =>  !(!value || value === 'false');

Not that familiar with the code, but even shorter & more readable code:
const toBoolean = (value: String): boolean =>  value === 'true';